tools/install_appveyor.py

```diff
@@ -0,0 +1,154 @@
+wget(r'https://github.com/bluescarni/binary_deps/raw/master/msgpack_mingw_64.7z', 'msgpack.7z')
+# Extract them.
+run_command(r'7z x -aoa -oC:\\ gmp.7z', verbose=False)
+run_command(r'7z x -aoa -oC:\\ mpfr.7z', verbose=False)
+run_command(r'7z x -aoa -oC:\\ boost.7z', verbose=False)
+run_command(r'7z x -aoa -oC:\\ msgpack.7z', verbose=False)
+
+# Set the path so that the precompiled libs can be found.
+os.environ['PATH'] = os.environ['PATH'] + r';c:\\local\\lib'
+
+# Build type setup.
+BUILD_TYPE = os.environ['BUILD_TYPE']
+is_master_build = (os.environ['APPVEYOR_REPO_BRANCH'] == 'master') and not os.environ['APPVEYOR_PULL_REQUEST_NUMBER']
+if is_master_build:
+    print("Master build detected.")
+is_python_build = 'Python' in BUILD_TYPE
+if is_python_build:
+    if BUILD_TYPE == 'Python35':
+        python_version = '35'
+    elif BUILD_TYPE == 'Python34':
+        python_version = '34'
+    elif BUILD_TYPE == 'Python27':
+        python_version = '27'
+    else:
+        raise RuntimeError('Unsupported Python build: ' + BUILD_TYPE)
+    python_package = r'python' + python_version + r'_mingw_64.7z'
+    boost_python_package = r'boost_python_' + python_version + r'_mingw_64.7z'
+    # Remove any existing Python installation.
+    rm_fr(r'c:\\Python' + python_version)
+    # Set paths.
+    pinterp = r'c:\\Python' + python_version + r'\\python.exe'
+    pip = r'c:\\Python' + python_version + r'\\scripts\\pip'
+    twine = r'c:\\Python' + python_version + r'\\scripts\\twine'
+    pyranha_install_path = r'C:\\Python' + \
+        python_version + r'\\Lib\\site-packages\\pyranha'
+    # Get Python.
+    wget(r'https://github.com/bluescarni/binary_deps/raw/master/' +
+         python_package, 'python.7z')
+    run_command(r'7z x -aoa -oC:\\ python.7z', verbose=False)
+    # Get Boost Python.
+    wget(r'https://github.com/bluescarni/binary_deps/raw/master/' +
+         boost_python_package, 'boost_python.7z')
+    run_command(r'7z x -aoa -oC:\\ boost_python.7z', verbose=False)
+    # Install pip and deps.
+    wget(r'https://bootstrap.pypa.io/get-pip.py', 'get-pip.py')
+    run_command(pinterp + ' get-pip.py')
+    run_command(pip + ' install numpy')
+    run_command(pip + ' install mpmath')
+    if is_master_build:
+        run_command(pip + ' install twine')
+
+# Proceed to the build.
+os.makedirs('build')
+os.chdir('build')
+
+common_cmake_opts = r'-DBoost_LIBRARY_DIR_RELEASE=c:\\local\\lib -DBoost_INCLUDE_DIR=c:\\local\\include -DGMP_INCLUDE_DIR=c:\\local\\include -DGMP_LIBRARIES=c:\\local\\lib\\libgmp.a -DMPFR_INCLUDE_DIR=c:\\local\\include -DMPFR_LIBRARIES=c:\\local\\lib\\libmpfr.a -DPIRANHA_WITH_BZIP2=yes -DBZIP2_INCLUDE_DIR=c:\\local\\include -DBZIP2_LIBRARY_RELEASE=c:\\local\\lib\\libboost_bzip2-mgw62-mt-1_62.dll -DPIRANHA_WITH_MSGPACK=yes -DMSGPACK-C_INCLUDE_DIR=c:\\local\\include -DPIRANHA_WITH_ZLIB=yes -DZLIB_INCLUDE_DIR=c:\\local\\include -DZLIB_LIBRARY_RELEASE=c:\\local\\lib\\libboost_zlib-mgw62-mt-1_62.dll'
```

Comment:
Ah good catch. I am not sure it will work for bzip2 and zlib though, as these are the boost-provided versions with non-standard names (e.g., ``libboost_bzip2-mgw62-mt-1_62.dll``). It should work for the rest though, I am gonna try now.